src/test/java/somethingrandom/dataaccess/local/AddItemTest.java

```diff
@@ -0,0 +1,143 @@
+            assertEquals(itemData.get("itemKind"), item1.getItemKind());
+            assertEquals(itemData.get("neededTime"), item1.getNeededTime().toString());
+        } catch (IOException e) {
+            throw new IOException(e);
+        }
+    }
+
+    @org.junit.Test
+    public void testDelayedItemSavedProperly() throws IOException {
+        try{
+            BufferedReader reader = new BufferedReader(new FileReader("./data.json"));
+            JSONObject data = new JSONObject(reader.readLine());
+            reader.close();
+            JSONObject item = data.getJSONObject(item2.getID().toString());
+            Map<String, String> itemData = new HashMap<>();
+
+            for (String itemKey : item.keySet()) {
+                if (!itemKey.equals("empty")) {
+                    itemData.put(itemKey, item.getString(itemKey));
+                }
+            }
+
+            assertEquals(itemData.get("name"), item2.getName());
+            assertEquals(itemData.get("id"), item2.getID().toString());
+            assertEquals(itemData.get("creationDate"), item2.getCreationDate().toString());
+            assertEquals(itemData.get("itemKind"), item2.getItemKind());
+            assertEquals(itemData.get("remindDate"), item2.getReminderDate().toString());
+        } catch (IOException e) {
+            throw new IOException(e);
+        }
+    }
+
+    @org.junit.Test
+    public void testReferenceItemSavedProperly() throws IOException {
+        try{
+            BufferedReader reader = new BufferedReader(new FileReader("./data.json"));
+            JSONObject data = new JSONObject(reader.readLine());
+            reader.close();
+            JSONObject item = data.getJSONObject(item3.getID().toString());
+            Map<String, String> itemData = new HashMap<>();
+
+            for (String itemKey : item.keySet()) {
+                if (!itemKey.equals("empty")) {
+                    itemData.put(itemKey, item.getString(itemKey));
+                }
+            }
+
+            assertEquals(itemData.get("name"), item3.getName());
+            assertEquals(itemData.get("id"), item3.getID().toString());
+            assertEquals(itemData.get("creationDate"), item3.getCreationDate().toString());
+            assertEquals(itemData.get("itemKind"), item3.getItemKind());
+            assertEquals(itemData.get("description"), item3.getDescription());
+        } catch (IOException e) {
+            throw new IOException(e);
+        }
+    }
+
+    @org.junit.Test
+    public void testDataPersists() throws IOException {
+        try{
+            ItemFactory factory = new CommonItemFactory();
+            FileUserDataAccessObject dao = new FileUserDataAccessObject("./data.json", factory);
+            testActionableItemSavedProperly();
+            testReferenceItemSavedProperly();
+            testDelayedItemSavedProperly();
+        } catch (IOException | DataAccessException e) {
+            throw new IOException(e);
+        }
```

Comment:
Does this test do anything the others don't?

Reply:
My goal with this test was to make sure that if I created a new dao on the existing file that contained the data, it would not over write it and delete the contents. I think I did it properly because if you comment out the load method and run the code, it passes all the test cases except `testDataPersists()`.

Reply:
OK, I think I was tripped up by the constructor doing I/O. Could add a comment explaining that, but otherwise good to go!